src/ints/mouse.cpp

```diff
@@ -31,21 +34,57 @@
 #include "bios.h"
 #include "dos_inc.h"
 
+using namespace bit::literals;
+
+MouseInfoConfig mouse_config;
+MouseInfoVideo mouse_video;
+
+enum class EventType : uint8_t { // compatible with DOS driver mask in driver function 0x0c
+	NotDosEvent    = 0x00,
+	MouseHasMoved  = 0x01,
+	PressedLeft    = 0x02,
+	ReleasedLeft   = 0x04,
+	PressedRight   = 0x08,
+	ReleasedRight  = 0x10,
+	PressedMiddle  = 0x20,
+	ReleasedMiddle = 0x40,
+	WheelHasMoved  = 0x80,
+};
+
+static uint8_t buttons_12  = 0; // state of buttons 1 (left), 2 (right), as visible on host side
+static uint8_t buttons_345 = 0; // state of mouse buttons 3 (middle), 4, and 5 as visible on host side
+
+MouseInfoConfig::MouseInfoConfig() : sensitivity_x(0.3f), sensitivity_y(0.3f) {}
+
+MouseInfoVideo::MouseInfoVideo()
+        : fullscreen(true),
+          res_x(320),
+          res_y(200),
+          clip_x(0),
+          clip_y(0)
+{}
+
+typedef struct MouseEvent {
+	uint8_t dos_type    = 0;
+	uint8_t dos_buttons = 0;
+
+	MouseEvent() {}
+	MouseEvent(uint8_t dos_type) : dos_type(dos_type) {}
+
+} MouseEvent;
+
+static constexpr uint8_t QUEUE_SIZE = 32; // if over 255, increase 'queue_used' size
+static MouseEvent queue[QUEUE_SIZE];
+static uint8_t    queue_used = 0;
+static bool       timer_in_progress = false;
+
 static Bitu call_int33,call_int74,int74_ret_callback,call_mouse_bd;
 static uint16_t ps2cbseg,ps2cbofs;
 static bool useps2callback,ps2callbackinit;
 static Bitu call_ps2,call_uir;
 static RealPt ps2_callback,uir_callback;
 static int16_t oldmouseX, oldmouseY;
```

Comment:
Now that quite wad of variables.

If you've got the stomach to clean these up, one-variable per-line, and assign defaults, it would help!

(Or feel free to ignore; happy for cleanup to happen in a future PR)

Reply:
I would prefer not to do this right now - indeed, some cleanup is needed here, and I already did it in my development branch, which I intend to submit as 'part 2'. See here:

https://github.com/FeralChild64/dosbox-staging/tree/fc/mouse-updates-2/src/ints